test/component_failures_test.py

```diff
@@ -299,8 +319,10 @@ def g(*xs):
     # execute. Do this in a loop while submitting tasks between each
     # component failure.
     time.sleep(0.1)
-    components = ray.services.all_processes[component_type]
-    for process in components[1:]:
+    worker_nodes = cluster.list_all_nodes()[1:]
```

Comment:
This is not great because it isn't an invariant provided/explicit behavior by `list_all_nodes`. It would be good to expose a specific method `list_worker_nodes()` instead of assuming the first one is the head.